crates/compiler/gen_llvm/src/llvm/sort.rs

```diff
@@ -0,0 +1,200 @@
+use super::build::BuilderExt;
+use crate::llvm::build::Env;
+use inkwell::values::{BasicValueEnum, IntValue};
+use inkwell::IntPredicate;
+use roc_builtins::bitcode::IntWidth;
+use roc_mono::layout::{
+    Builtin, InLayout, LayoutIds, LayoutInterner, LayoutRepr, STLayoutInterner,
+};
+
+pub fn generic_compare<'a, 'ctx>(
+    env: &Env<'a, 'ctx, '_>,
+    layout_interner: &STLayoutInterner<'a>,
+    _layout_ids: &mut LayoutIds<'a>,
+    lhs_val: BasicValueEnum<'ctx>,
+    rhs_val: BasicValueEnum<'ctx>,
+    lhs_layout: InLayout<'a>,
+    _rhs_layout: InLayout<'a>,
+) -> BasicValueEnum<'ctx> {
+    let lhs_repr = layout_interner.get_repr(lhs_layout);
+    match lhs_repr {
+        LayoutRepr::Builtin(Builtin::Int(int_width)) => {
+            int_compare(env, lhs_val, rhs_val, int_width)
+        }
+        LayoutRepr::Builtin(Builtin::Float(_)) => todo!(),
+        LayoutRepr::Builtin(Builtin::Bool) => todo!(),
+        LayoutRepr::Builtin(Builtin::Decimal) => todo!(),
+        LayoutRepr::Builtin(Builtin::Str) => todo!(),
+        LayoutRepr::Builtin(Builtin::List(_)) => todo!(),
+        LayoutRepr::Struct(_) => todo!(),
+        LayoutRepr::LambdaSet(_) => unreachable!("cannot compare closures"),
+        LayoutRepr::FunctionPointer(_) => unreachable!("cannot compare function pointers"),
+        LayoutRepr::Erased(_) => unreachable!("cannot compare erased types"),
+        LayoutRepr::Union(_) => todo!(),
+        LayoutRepr::Ptr(_) => todo!(),
+        LayoutRepr::RecursivePointer(_) => todo!(),
+    }
+}
+
+fn int_compare<'ctx>(
+    env: &Env<'_, 'ctx, '_>,
+    lhs_val: BasicValueEnum<'ctx>,
+    rhs_val: BasicValueEnum<'ctx>,
+    builtin: IntWidth,
+) -> BasicValueEnum<'ctx> {
+    // The following calculation will return 0 for equals, 1 for greater than,
+    // and 2 for less than.
+    // (a > b) + 2 * (a < b);
```

Comment:
the reason is that this will map directly to the variants that roc would generate for the ordering type, right?

(picking -1, 0, 1 would be more standard in general, but for us would make less sense because of the above)

maybe these functions should return `IntValue` to make the idea clearer

Reply:
> the reason is that this will map directly to the variants that roc would generate for the ordering type, right?

Yes! Thanks for the tip, I'll make them return `IntValue`.